container-search/src/main/java/com/yahoo/search/yql/YqlParser.java

```diff
@@ -1329,8 +1331,8 @@ private Item instantiateWordItem(String field,
     }
 
     @SuppressWarnings({"deprecation"})
-    private boolean shouldSegment(String field, boolean fromQuery) {
-        return fromQuery && ! indexFactsSession.getIndex(field).isAttribute();
+    private boolean shouldSegment(String field, OperatorNode<ExpressionOperator> ast, boolean fromQuery) {
```

Comment:
why do you take the "ast" as a parameter here?

Reply:
To see if you are paying attention ... duly impressed.

Also, I thought I could use the getIndex(ast) method, which seems to do the right thing, but that makes further, not necessarily true assumptions. When this is safely out, I'm going to have another pass at this code to make it more understandable, so I didn't bother to clean up that parameter now.